client/components/common/SelectElement/index.vue

```diff
@@ -2,102 +2,212 @@
+          :content-containers="items.contentContainers"
+          :selected="selection.elements"
+          :allowed-types="allowedTypes"
+          :multiple="multiple"
+          selectable />
+      </template>
     </template>
     <template v-slot:actions>
       <v-btn
-        v-if="selectedActivity"
-        @click="selectedActivity = null"
+        v-if="selection.activity"
+        @click="deselectActivity"
         text outlined
         class="mr-2">
         <v-icon>mdi-arrow-left</v-icon> Back
       </v-btn>
       <v-btn @click="close" text class="ml-1">Cancel</v-btn>
-      <v-btn @click="save" color="secondary" text>Save</v-btn>
+      <v-btn @click="save" color="secondary" text>{{ submitLabel }}</v-btn>
     </template>
   </tailor-dialog>
 </template>
 
 <script>
+import activitiesApi from '@/api/activity';
 import contentElementApi from 'client/api/contentElement';
 import ContentPreview from '@/components/common/ContentPreview';
+import flatMap from 'lodash/flatMap';
 import { getDescendants as getContainers } from '@/utils/activity';
+import { getSupportedContainers } from 'shared/activities';
 import loader from '@/components/common/loader';
+import map from 'lodash/map';
 import { mapGetters } from 'vuex';
 import SelectActivity from './SelectActivity';
+import SelectRepository from './SelectRepository';
 import sortBy from 'lodash/sortBy';
 import TailorDialog from '@/components/common/TailorDialog';
 
+const getContainerTypes = type => map(getSupportedContainers(type), 'type');
+
+const TOGGLE_BUTTON = {
+  SELECT: { label: 'Select all', icon: 'checkbox-multiple-marked-outline' },
+  DESELECT: { label: 'Deselect all', icon: 'checkbox-multiple-blank-outline' }
+};
+
 export default {
   name: 'select-element',
   props: {
     selected: { type: Array, default: () => [] },
     heading: { type: String, required: true },
+    allowedTypes: { type: Array, required: true },
     multiple: { type: Boolean, default: true },
-    allowedTypes: { type: Array, default: () => [] }
+    submitLabel: { type: String, default: 'Save' },
+    headerIcon: { type: String, default: 'mdi-toy-brick-plus-outline' },
+    onlyCurrentRepo: { type: Boolean, default: false }
   },
   data: () => ({
-    selectedActivity: null,
-    contentContainers: [],
-    selectedElements: [],
+    items: {
+      activities: [],
+      contentContainers: []
+    },
+    selection: {
+      repository: null,
+      activity: null,
+      elements: []
+    },
     loadingContent: false
   }),
-  computed: mapGetters('repository', ['repository', 'activities']),
+  computed: {
+    ...mapGetters('repository', {
+      currentRepository: 'repository',
+      currentActivities: 'activities'
+    }),
+    allElementsSelected: vm => vm.selection.elements.length === vm.elements.length,
+    rootContainerTypes() {
+      const type = this.selection.activity?.type;
+      return type && getContainerTypes(type);
+    },
+    processedContainers() {
```

Comment:
It might be just a function (we have already a lot of `something` containers on the context. It is only used within a single function so it can be eliminated; Separate PR)